copilot/fsd-account-store/addons/fsd-account-store-cluster.yml

```diff
@@ -0,0 +1,140 @@
+    Type: 'AWS::RDS::DBClusterParameterGroup'
+    Properties:
+      Description: !Ref 'AWS::StackName'
+      Family: 'aurora-postgresql14'
+      Parameters:
+        client_encoding: 'UTF8'
+  fsdaccountstoreclusterDBCluster:
+    Metadata:
+      'aws:copilot:description': 'The fsdaccountstorecluster Aurora Serverless v2 database cluster'
+    Type: 'AWS::RDS::DBCluster'
+    Properties:
+      MasterUsername:
+        !Join [ "",  [ '{{resolve:secretsmanager:', !Ref fsdaccountstoreclusterAuroraSecret, ":SecretString:username}}" ]]
+      MasterUserPassword:
+        !Join [ "",  [ '{{resolve:secretsmanager:', !Ref fsdaccountstoreclusterAuroraSecret, ":SecretString:password}}" ]]
+      DatabaseName: !Ref fsdaccountstoreclusterDBName
+      Engine: 'aurora-postgresql'
+      EngineVersion: '14.4'
+      DBClusterParameterGroupName: !Ref fsdaccountstoreclusterDBClusterParameterGroup
+      DBSubnetGroupName: !Ref fsdaccountstoreclusterDBSubnetGroup
+      Port: 5432
+      VpcSecurityGroupIds:
+        - !Ref fsdaccountstoreclusterDBClusterSecurityGroup
+      ServerlessV2ScalingConfiguration:
+        # Replace "All" below with "!Ref Env" to set different autoscaling limits per environment.
+        MinCapacity: !FindInMap [fsdaccountstoreclusterEnvScalingConfigurationMap, All, DBMinCapacity]
+        MaxCapacity: !FindInMap [fsdaccountstoreclusterEnvScalingConfigurationMap, All, DBMaxCapacity]
+  fsdaccountstoreclusterDBWriterInstance:
+    Metadata:
+      'aws:copilot:description': 'The fsdaccountstorecluster Aurora Serverless v2 writer instance'
+    Type: 'AWS::RDS::DBInstance'
+    Properties:
+      DBClusterIdentifier: !Ref fsdaccountstoreclusterDBCluster
+      DBInstanceClass: db.serverless
+      Engine: 'aurora-postgresql'
+      PromotionTier: 1
+      AvailabilityZone: !Select
+        - 0
+        - !GetAZs
+          Ref: AWS::Region
+
+  fsdaccountstoreclusterSecretAuroraClusterAttachment:
+    Type: AWS::SecretsManager::SecretTargetAttachment
+    Properties:
+      SecretId: !Ref fsdaccountstoreclusterAuroraSecret
+      TargetId: !Ref fsdaccountstoreclusterDBCluster
+      TargetType: AWS::RDS::DBCluster
+Outputs:
+  DatabaseUrl:
+    Description: "The URL of this database."
+    Value:
+      !Sub
+      - "postgres://${USERNAME}:${PASSWORD}@${HOSTNAME}:${PORT}/${DBNAME}"
+      - USERNAME: !Join [ "",  [ '{{resolve:secretsmanager:', !Ref fsdaccountstoreclusterAuroraSecret, ":SecretString:username}}" ]]
+        PASSWORD: !Join [ "",  [ '{{resolve:secretsmanager:', !Ref fsdaccountstoreclusterAuroraSecret, ":SecretString:password}}" ]]
+        HOSTNAME: !Join [ "",  [ '{{resolve:secretsmanager:', !Ref fsdaccountstoreclusterAuroraSecret, ":SecretString:host}}" ]]
+        PORT: !Join [ "",  [ '{{resolve:secretsmanager:', !Ref fsdaccountstoreclusterAuroraSecret, ":SecretString:port}}" ]]
+        DBNAME: !Join [ "",  [ '{{resolve:secretsmanager:', !Ref fsdaccountstoreclusterAuroraSecret, ":SecretString:dbname}}" ]]
```

Comment:
I do like the way this turned out.  Very easy to understand.

Reply:
It's a much better portmanteau than the join of a join of a join of a join that I was initially writing, yes